@profxj @jhennawi I did some tests.
I run the Dev Suite and passed. 
```
--- PYPEIT DEVELOPMENT SUITE PASSED 102/102 TESTS (Masters ignored) ---
Testing Started at 2021-04-18T16:54:17.515374
Testing Completed at 2021-04-18T23:03:57.172152
Total Time: 6:09:39.656778
```

 I also run the unit tests but one test did not pass
`FAILED test_slitmask.py::test_add_missing_obj`

The reason it did not pass is because the test is reading in a cooked spec1d, which goes through `mask_add_missing_obj` (where `vette_for_extraction` is run). Because there are SpecObj keys that are not saved in the output datamodel ('trace_spec' and 'maskwidth'), `vette_for_extraction` does not find them.

So, do we want to include these ('trace_spec' and 'maskwidth') in the output datamodel (as suggested by @jhennawi ) or should I just fix the unit test?

--- a/pypeit/specobj.py
+++ b/pypeit/specobj.py
@@ -153,6 +153,10 @@ class SpecObj(datamodel.DataContainer):
                                      descr='Spatial location of the trace on detector (pixel) at half-way'),
                  'SPAT_FRACPOS': dict(otype=(float, np.floating),
                                       descr='Fractional location of the object on the slit'),
+                 'trace_spec': dict(otype=np.ndarray, atype=(int,np.integer),
+                                      descr='Array of pixels along the spectral direction'),
+                 'maskwidth': dict(otype=(float, np.floating),
+                                      descr='Size (in units of fwhm) of the region used for local sky subtraction'),
                  # Slit and Object
                  'WAVE_RMS': dict(otype=(float, np.floating),
                                      descr='RMS (pix) for the wavelength solution for this slit.'),
@@ -576,6 +580,7 @@ class SpecObj(datamodel.DataContainer):
         for key in required:
             if self[key] is None:
                 msgs.warn("Item {} is missing from SpecObj. Failing vette".format(key))
+                msgs.warn('{}'.format(self))
                 passed = False
         #
         return passed

